app/src/main/java/com/example/android/psiak/FirebaseActivity.java

```diff
@@ -0,0 +1,158 @@
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class FirebaseActivity
+        extends AppCompatActivity
+        implements FirebaseActivityContract.View {
+
+    private static final String TAG = FirebaseActivity.class.toString();
+
+    // region Properties
+
+    private FirebasePresenter firebasePresenter;
+
+    @BindView(R.id.btn_show_all_dogs)
+    Button btnShowAllDogs;
+    @BindView(R.id.btn_add_new_dog)
+    Button btnAddNewDog;
+
+    @BindView(R.id.et_dog_name)
+    EditText etDogName;
+    @BindView(R.id.et_dog_gender)
+    EditText etDogGender;
+    @BindView(R.id.et_dog_age)
+    EditText etDogAge;
+    @BindView(R.id.et_dog_description)
+    EditText etDogDescription;
+    @BindView(R.id.et_dog_size)
+    EditText etDogSize;
+    @BindView(R.id.et_dog_weight)
+    EditText etDogWeight;
+    @BindView(R.id.et_dog_location)
+    EditText etDogLocation;
+    @BindView(R.id.et_dog_attitude_people)
+    EditText etDogAttitudePeople;
+    @BindView(R.id.et_dog_attitude_dogs)
+    EditText etDogAttitudeDogs;
+    @BindView(R.id.et_dog_attitude_cats)
+    EditText etDogAttitudeCats;
+    @BindView(R.id.et_dog_keeper_name)
+    EditText etDogKeeperName;
+    @BindView(R.id.et_dog_keeper_phone)
+    EditText etDogKeeperPhone;
+    @BindView(R.id.et_dog_keeper_mail)
+    EditText etDogKeeperMail;
+    @BindView(R.id.et_dog_homeless_since)
+    EditText etDogHomelessSince;
+    @BindView(R.id.sw_dog_vaccinated)
+    Switch swDogVaccinated;
+    @BindView(R.id.sw_dog_dewormed)
+    Switch swDogDewormed;
+    @BindView(R.id.sw_dog_sterilized)
+    Switch swDogSterilized;
+
+    // endregion
+
+    // region Activity Lifecycle
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_firebase_testing);
+        ButterKnife.bind(this);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        // TODO Use dependency injection here
+        firebasePresenter = new FirebasePresenter(new FirebaseRepository());
+        firebasePresenter.attach(this);
+
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        if (firebasePresenter != null)
+            firebasePresenter.detach();
+    }
+
+    // endregion
+
+    // region Private Methods
+
+    /**
+     * Get all Dog's objects form database
+     */
+
+    @OnClick(R.id.btn_show_all_dogs)
+    void getAllDogs(View view) {
+        firebasePresenter.getAllDogs();
+    }
+
+    /**
+     * Add new dog to the Firebase database
+     */
+
+    @OnClick(R.id.btn_add_new_dog)
+    void addNewDogDog(View view) {
+
+        String uniqueID = firebasePresenter.generateUniqueID();
+
+        DogFirebase dogFirebase = new DogFirebase.DogBuilder(uniqueID, etDogName.getText().toString())
```

Comment:
I'm open to suggestions, because I don't know to change this code.

Reply:
Well to be honest I do not know any proper way to do so, so we can leave it for now as it is.
I opened thread on Udacity, so maybe someone suggest us best way to do so. :)

Reply:
From Udacity: a simple way of dealing with this would be to write a simple function that takes the edittext as a parameter and returns it’s text via getText().toString()